clang/include/clang/CIR/Dialect/Builder/CIRBaseBuilder.h

```diff
@@ -396,9 +396,25 @@ class CIRBaseBuilderTy : public mlir::OpBuilder {
 
   mlir::Value createGetGlobal(mlir::Location loc, cir::GlobalOp global,
                               bool threadLocal = false) {
-    return create<cir::GetGlobalOp>(
+    auto getGlobal = create<cir::GetGlobalOp>(
         loc, getPointerTo(global.getSymType(), global.getAddrSpaceAttr()),
         global.getName(), threadLocal);
+
+    // When AST attribute is not present, the global is a temporary,
```

Comment:
This connection between presence of the AST and global temporaries seems a bit confusing. Perhaps just say "Check whether actual & expected address match. Use AST convenience to retrieve more information"